### PR TITLE
Vue route based menu config

### PR DESCRIPTION
**Pull request by d-koppenhagen**

### CODE REVIEW

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+<template>
+  <div>MainPage.vue</div>
+</template>
+```
+
+The next thing we want to do is to setup the route for this component.
+Therefore we are creating the `router.ts` file within the `src` directory.
+We are importing the `MainPage` component and using it for the route `main`.
+Furthermore we are adding an redirect to "`/main`" when the root-route "`/`" is opened.
```
Comment (jschirrmacher) [inline blog/2022-10-vue-dynamic-menu-route-based.md:72]:
```suggestion
Furthermore we are adding a redirect to "`/main`" when the root-route "`/`" is opened.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    component: MainPage,
+    meta: {
+      label: 'Main Page',
+    },
+  }
+];
+
+export const router = createRouter({
+  history: createWebHistory(),
+  routes: [...routes, ...pluginRoutes.flat()],
```
Comment (jschirrmacher) [inline blog/2022-10-vue-dynamic-menu-route-based.md:96]: Where does `pluginRoutes` come from? Can we leave it out from this example?

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+<template>
+  <AppNav />
+  <hr />
+  <router-view />
+</template>
+```
+
+Next, we are creating the `NavItem` component.
+We are defineing a single prop it get's passed by the parent component called `routeConfig` that contains a whole route configuration record.
```
Comment (jschirrmacher) [inline blog/2022-10-vue-dynamic-menu-route-based.md:173]:
```suggestion
We are defineing a single prop which gets passed by the parent component called `routeConfig` that contains a whole route configuration record.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+Now we can focus on the template.
+We are adding a `<router-link>` and we will pass the route target by using the unique `name`.
+As the label of the link we can now us the `label` from our `meta` information object we provided in the router configuration.
```
Comment (jschirrmacher) [inline blog/2022-10-vue-dynamic-menu-route-based.md:176]:
```suggestion
As the label of the link we can now use the `label` from our `meta` information object we provided in the router configuration.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+
+Great! The hardest part is finished now (wasn't that tricky right?) and probably this solution already fit's for a majority.
+But two things I want to describe in advance as they may be relevant for you:
+
+1. How to make the navigation easily extensible
+2. How to implement child menu items
+
+### Make the navigation extensible
+
+Let's assume we have an extensible app where we will outsource some whole pages and it's child route configurations and make them incluable in our app.
```
Comment (jschirrmacher) [inline blog/2022-10-vue-dynamic-menu-route-based.md:205]:
```suggestion
Let's assume we have an extensible app where we will outsource some some pages and it's child route configurations and make them includable in our app.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+Therefore we want to make our route configuration extensible, so that we can pass additional routes and child routes linked with their components to our router.
+
+To do so, we simply wrap the exported route into a function that accepts a list of route configurations as a Rest parameter.
+
+```ts
+/* ... */
+export function getConfiguredRouter(...pluginRoutes: RouteRecordRaw[][]) {
+  return createRouter({
+    history: createWebHistory(),
+    routes: [...routes, ...pluginRoutes.flat()],
```
Comment (jschirrmacher) [inline blog/2022-10-vue-dynamic-menu-route-based.md:217]: Now `pluginRoutes` make sense!

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+  background-color: lightblue;
+}
+a.router-link-exact-active {
+  background-color: #f3aff8;
+}
+```
+
+## Conclusion
+
+Passing meta information like `label` to the vue router configuration let's us easily building dynamic generated menus.
```
Comment (jschirrmacher) [inline blog/2022-10-vue-dynamic-menu-route-based.md:320]:
```suggestion
Passing meta information like `label` to the vue router configuration lets us easily build dynamic generated menus.
```

**Review by jschirrmacher — APPROVED** [review]

**Review by d-koppenhagen — COMMENTED** [review]
fix routes

File: blog/2022-10-vue-dynamic-menu-route-based.md (hunk 5 above)
Comment (d-koppenhagen) [reply blog/2022-10-vue-dynamic-menu-route-based.md:205]:
```suggestion
Let's assume we have an extensible app where we will outsource some pages and it's child route configurations and make them includable in our app.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 8]
```diff
@@ -0,0 +1,330 @@
...
+2. How to implement child menu items
+
+### Make the navigation extensible
+
+Let's assume we have an extensible app where we outsource some pages and its child route configurations and make them includable in our app.
+This could for example be relevent when adding complete menus and pages for specific users with appropriate permissions.
+
+Therefore we want to make our route configuration extensible, so we can pass additional routes and child routes linked with their components to our router.
+
+To do this, we simply wrap the exported route into a function that accepts a list of route configurations as a Rest parameter.
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:210]:
```suggestion
To do this, we simply move the exported route into a function that accepts a list of route configurations as a Rest parameter.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 9]
```diff
@@ -0,0 +1,330 @@
...
+---
+
+
+# Build a dynamic navigation based on route configuration using Vue3 and Vue Router
+
+Recently while working on [Vue](https://vuejs.org/) app, I asked myself: Isn’t the main navigation menu somehow related to the configuration of the routes and routing tree? And can't it be built dynamically from the router configuration?
+
+With this question in my mind, I started to work on a very simple but representative example of how to achieve this by enriching the route configuration using the `meta` option.
+
+The following example allows you to easily wrap big parts of your app into a module that is self contained and only exposes a bit of route configuration which can be imported and included in the main router configuration.
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:29]:
```suggestion
The following example allows you to easily place big parts of your app into a module that is self contained and only exposes a bit of route configuration which can be imported and included in the main router configuration.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 10]
```diff
@@ -0,0 +1,330 @@
+---
+title: 'Route based navigation menus in Vue'
+description: 'Learn how to build a dynamic navigation menu based on the route configuration using Vue3 and Vue Router'
+published: true
+author:
+  name: 'Danny Koppenhagen'
+  mail: mail@k9n.dev
+created: 2022-10-15
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:8]: Todo: update before publish

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 11]
```diff
@@ -0,0 +1,330 @@
...
+keywords:
+  - Vue
+  - Vue 3
+  - Vue Router
+language: en
+thumbnail:
+  header: assets/images/blog/vue-route-menu.jpg
+  card: assets/images/blog/vue-route-menu-small.jpg
+linked:
+  devTo: ''
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:19]: Todo: cross post at dev.to and link it here

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 12]
```diff
@@ -0,0 +1,330 @@
...
+---
+
+
+# Build a dynamic navigation based on route configuration using Vue3 and Vue Router
+
+Recently while working on [Vue](https://vuejs.org/) app, I asked myself: Isn’t the main navigation menu somehow related to the configuration of the routes and routing tree? And can't it be built dynamically from the router configuration?
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:25]:
```suggestion
Recently while working on a [Vue](https://vuejs.org/) app, I asked myself: Isn’t the main navigation menu somehow related to the configuration of the routes and routing tree? And can't it be built dynamically from the router configuration?
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 13]
```diff
@@ -0,0 +1,330 @@
...
+</script>
+
+<template>
+  <AppNav />
+  <hr />
+  <router-view />
+</template>
+```
+
+Next, we are create the `NavItem` component.
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:172]:
```suggestion
Next, we create the `NavItem` component.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 14]
```diff
@@ -0,0 +1,330 @@
...
+We are defining a single prop which gets passed by the parent component called `routeConfig` which contains a whole route configuration record.
+Now we can focus on the template.
+Next add a `<router-link>` and pass the route target using the unique `name`.
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:175]:
```suggestion
Now we can focus on the template:
Add a `<router-link>` and pass the route target using the unique `name`.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 15]
```diff
@@ -0,0 +1,330 @@
...
+        </router-link>
+      </li>
+    </ul>
+  </li>
+</template>
+```
+
+## Styling active links 
+
+To highlight the active menu items, we can now use the two automatically create CSS classes `router-link-active` and `router-link-exact-active`.
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:304]:
```suggestion
To highlight the active menu items, we can now use the two automatically created CSS classes `router-link-active` and `router-link-exact-active`.
```

File: blog/2022-10-vue-dynamic-menu-route-based.md [hunk 16]
```diff
@@ -0,0 +1,330 @@
...
+We no longer have to manually adjust our main navigation when adding new sites to our page as the menu is automatically extended by accessing the routes `meta` information.
+This approach can reduce some template boilerplate.
+
+You can use this approach to loosely couple whole parts of your app by adding them as separate modules without the need to add internals like the navigation titles to the main app part.
+
+The whole working example can be seen in the following Stackblitz project:
+
+https://stackblitz.com/edit/vue3-dynamic-menu
+
+> Thanks for ??? for reviewing this article.
```
Comment (d-koppenhagen) [inline blog/2022-10-vue-dynamic-menu-route-based.md:330]: Todo: Add Darren and Joachim

**Pull request merged**